crypto/provider_core.c

```diff
@@ -733,14 +738,14 @@ int ossl_provider_activate(OSSL_PROVIDER *prov, int retain_fallbacks)
             prov->store->use_fallbacks = 0;
             CRYPTO_THREAD_unlock(prov->store->lock);
         }
-        return 1;
+        return provider_flush_store_cache(prov);
```

Comment:
This flushing seems too aggressive. I think we only need to flush if the `activatecnt` drops to 0, or gets incremented to 1.

Reply:
I guess so...